### PR TITLE
Lint and exponential backoff duration fixes.

### PR DESCRIPTION
**Pull request by abbychau**
This PR fixes a minor bug on potential incorrect factor of duration in backoff;
and some vet issues.

### CODE REVIEW

File: network/network.go [hunk 1]
```diff
@@ -522,7 +522,6 @@ func (n *Network) Write(address string, message *protobuf.Message) error {
 		return errors.Errorf("worker must be too busy; failed to send message to %s", address)
 	}
 
-	return nil
```
Comment (iwasaki-kenta) [inline network/network.go]: Why remove `return nil` here?

File: examples/proxy/proxy_test.go [hunk 2]
```diff
@@ -90,11 +90,11 @@ func (n *ProxyPlugin) ProxyBroadcast(node *network.Network, sender peer.ID, msg
 	return nil
 }
 
-// ExampleProxy demonstrates how to send a message to nodes which do not directly have connections
+// ProxyExample demonstrates how to send a message to nodes which do not directly have connections
 // to their desired messaging target.
 //
 // Messages are proxied to closer nodes using the Kademlia routing table.
-func ExampleProxy() {
+func ProxyExample() {
```
Comment (iwasaki-kenta) [inline examples/proxy/proxy_test.go]: Best keep the prefix as `Example` to stick to Go's recommendations for naming.

File: examples/basic/example_test.go [hunk 3]
```diff
@@ -30,9 +30,9 @@ func (state *BasicPlugin) Receive(ctx *network.PluginContext) error {
 	return nil
 }
 
-// ExampleBasic demonstrates how to broadcast a message to a set of peers that discover
+// BasicExample demonstrates how to broadcast a message to a set of peers that discover
 // each other through peer discovery.
-func ExampleBasic() {
+func BasicExample() {
```
Comment (iwasaki-kenta) [inline examples/basic/example_test.go]: Best keep the prefix as `Example` to stick to Go's recommendations for naming.

File: network/network.go (hunk 1 above)
Comment (abbychau) [reply network/network.go]: it is unreachable.

File: examples/proxy/proxy_test.go (hunk 2 above)
Comment (abbychau) [reply examples/proxy/proxy_test.go]: go vet is assuming XXXExample with an identifier named "XXX".
actually the recommendation makes this naming not good. it is a demonstration of something but not an example of an identifier.

Comment (iwasaki-kenta) [reply examples/proxy/proxy_test.go]: https://blog.golang.org/examples That's where we got the recommendation from.

Comment (abbychau) [reply examples/proxy/proxy_test.go]: um ... `ExampleProxyPlugin` could handle both cases.

Comment (iwasaki-kenta) [reply examples/proxy/proxy_test.go]: Sure, sounds good 👍 Also need to rename `BasicExample` as well 😄.

Comment (abbychau) [reply examples/proxy/proxy_test.go]: done~

**Pull request merged**